### PR TITLE
Add sign in with Apple to the sample app

### PR DESCRIPTION
**Pull request by renkelvin**

### CODE REVIEW

File: Example/Auth/Sample/MainViewController+Apple.m [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+#import "AppManager.h"
+#import "FirebaseAuth.h"
+#import "MainViewController+Internal.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MainViewController () <ASAuthorizationControllerDelegate, ASAuthorizationControllerPresentationContextProviding> {
+
+}
```
Comment (morganchen12) [inline Example/Auth/Sample/MainViewController+Apple.m]: nit: empty `{ }` can be removed

File: Example/Auth/Sample/MainViewController+Apple.m [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    [StaticContentTableViewCell cellWithTitle:@"Reauthenticate with Apple"
+                                       action:^{ [weakSelf reauthenticateWithApple]; }],
+  ]];
+}
+
+- (void)signInWithApple API_AVAILABLE(ios(13.0)) {
+  ASAuthorizationAppleIDProvider* provider = [[ASAuthorizationAppleIDProvider alloc] init];
+  ASAuthorizationAppleIDRequest* request = [provider createRequest];
+  request.requestedScopes = @[ASAuthorizationScopeEmail, ASAuthorizationScopeFullName];
+  request.nonce = @"c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2";
```
Comment (morganchen12) [inline Example/Auth/Sample/MainViewController+Apple.m]: This is not great nonce practice, maybe add a comment telling users to not do this in production.

Also, consider deduplicating the request config boilerplate into a separate method. Something like
```objc
- (ASAuthorizationAppleIDRequest *)appleIDRequestWithState:(NSString *)state {
  ASAuthorizationAppleIDRequest *request = [[[ASAuthorizationAppleIDProvider alloc] init] createRequest];
  request.requestedScopes = @[ ... ];
  request.nonce = @"";
  request.state = state;
  return request;
}
```

Comment (renkelvin) [reply Example/Auth/Sample/MainViewController+Apple.m]: Yeah, good idea.

**Comment by thejohnlima** [comment]
@renkelvin it seems in your latest commit the CI starts fail

**Comment by renkelvin** [comment]
> @renkelvin it seems in your latest commit the CI starts fail

Yeah, there is a weird CI issue.

**Pull request closed without merging**